openvdb/tools/GridTransformer.h

```diff
@@ -267,14 +267,24 @@ class GridTransformer: public GridResampler
 
 namespace local_util {
 
-/// @brief Decompose an affine transform into scale, rotation and translation components.
-/// @return @c false if the given matrix is not affine or cannot otherwise be decomposed.
+enum { DECOMP_INVALID = 0, DECOMP_VALID = 1, DECOMP_UNIQUE = 2 };
+
+/// @brief Decompose an affine transform into scale, rotation (XYZ order),
+/// and translation components.
+/// @return DECOMP_INVALID if the given matrix is not affine or cannot
+/// be decomposed, DECOMP_UNIQUE if the matrix has a unique decomposition,
+/// DECOMP_VALID otherwise
 template<typename T>
-inline bool
+inline int
 decompose(const math::Mat4<T>& m, math::Vec3<T>& scale,
     math::Vec3<T>& rotate, math::Vec3<T>& translate)
 {
-    if (!math::isAffine(m)) return false;
+    if (!math::isAffine(m)) return DECOMP_INVALID;
+
+    const auto posAngle = [](T angle) {
```

Comment:
can you explain or motivate this functor in a brief comment. It's certainly not self explanatory to me. It seems to be replace an abs operation further down in the code - was that a bug?

Reply:
My understanding of the code as a whole:  when you have a reflection, you have the issue of how to apply it - it could be a reflection through the origin, or it could be a plane reflection + 180 degree turn.   In Houdini I know we always select reflection through the origin as the most unbiased choice, but it is also rarely the one people "mean".

The code seems to be looking for the euler angles with the minimial change from identity to find the best choice of plane reflections to introduce.

This function seems to be doing a poor-man's modulus, understandable since C++ has a poorly defined modulus for negative numbers.    However, I'm not sure that's what one wants if one wants minimal rotation.  A slight negative rotation is going to return M_PI*2 so flag as being a large rotation, so not acceptable as a solution.

I think the abs() version should be kept instead?